x-pack/test/cases_api_integration/security_and_spaces/tests/common/internal/bulk_get_cases.ts

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { pick } from 'lodash';
+import expect from '@kbn/expect';
+import { CommentType } from '@kbn/cases-plugin/common';
+import { getPostCaseRequest, postCaseReq } from '../../../../common/lib/mock';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import {
+  deleteAllCaseItems,
+  bulkGetCases,
+  createCase,
+  createComment,
+  ensureSavedObjectIsAuthorized,
+} from '../../../../common/lib/utils';
+import {
+  secOnly,
+  obsOnly,
+  superUser,
+  secOnlyRead,
+  obsOnlyRead,
+  obsSecRead,
+  globalRead,
+  noKibanaPrivileges,
+  obsSec,
+} from '../../../../common/lib/authentication/users';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext): void => {
+  const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const es = getService('es');
+
+  describe('bulk_get_cases', () => {
+    afterEach(async () => {
+      await deleteAllCaseItems(es);
+    });
+
+    describe('get', () => {
+      it('should return the correct cases', async () => {
+        const caseOne = await createCase(supertest, postCaseReq);
+        const caseTwo = await createCase(supertest, postCaseReq);
+
+        const { cases, errors } = await bulkGetCases({ supertest, ids: [caseOne.id, caseTwo.id] });
+
+        expect(cases).to.eql([caseOne, caseTwo]);
+        expect(errors.length).to.be(0);
+      });
+
+      it('should return the correct cases with specific fields', async () => {
```

Comment:
`totalComment` and `totalAlerts` are handled slightly differently, could we get a test for when those two are the fields specified?

Reply:
There is a test for them below called `should return the correct counts`. Do you want me to add something else?

Reply:
But there they aren't passed in the fields params.

I was thinking of testing the snippet below

```
    const requestForTotals = ['totalComment', 'totalAlerts'].some(
      (totalKey) => !fields || fields.includes(totalKey)
    );
```

 but I just noticed you do `!fields ||`.

So that test should be enough, no worries 👍

Reply:
Thanks!